### ISSUE TITLE
question and bug report

### ISSUE THREAD

**Issue by lila** [gc3-uzh-ch/elasticluster#393]
I am using elasticluster with GCE to build slurm cluster.  Below are some issues that I had that seem to be due to debian 8 support.  What OS/distro should i use that has the best support?

1. dbus package missing on debian 8 vanilla distribution.  I added 

```
image_userdata=#!/bin/bash
  apt-get update && apt-get install -y dbus
```

to the config, but it seems that the GCE plugin scripts ignore the userdata and do not use it when spinning up the instance.  so you have to manually log in and install dbus.

in gce.py:

```           
"metadata": {
                "kind": "compute#metadata",
                "items": [
                    {
                        "key": "sshKeys",
                        "value": "%s:%s" % (username, public_key_content)
                    }
                ]
            }
        }
```

should have the following added to the array of items:

```
{
        # Startup script is automatically executed by the
        # instance upon startup.
        "key": "startup-script",
         "value": image_userdata
   }
```

2. even though i'm running debian 8, there are many places where the playbook directive is_debian_8_or_later seems to be inconsistent.

for example, in `elasticluster/share/playbooks/roles/slurm-client/tasks/main.yml` or in `elasticluster/share/playbooks/roles/slurm-master/tasks/init-Debian.yml`.  looking at the latter:

```
- name: Set SLURM master playbook params (Debian/Ubuntu)
  set_fact:
    slurmctld_service_name: slurmctld
    slurmctld_packages:
      - mailutils
      - slurmctld
    slurmdbd_service_name: slurmdbd
    slurmdbd_packages:
      - slurmdbd
  when:
    '{{is_debian_8_or_later}} or {{is_ubuntu_15_10_or_later}}'
```

does not set the slurmbd packages.  if i delete the when clause it's fine.  the `is_debian_8_or_later` clause seems to be inconsistently set.  sometimes its correct, sometimes its not.  

any advice here?

or in general, what would you all recommend for the best supported distribution?

cheers,

- k

**Comment by lila** [comment]
with the fix adding image_userdata to the gce plugin, i get another error since apt runs in the startup script as well as in the playbooks.  the playbook installation fails.

**Comment by riccardomurri** [comment]
hello @lila -- you reported a number of different problems (each of which would likely be worth its own thread) , then you closed the issue youself?  Did you find a way to solve them?  If yes, it would be nice to also hint at the solution here so others can maybe profit, or I can incorporate the fix into upstream ElastiCluster.  If not, would you mind reporting them as separate issues?

**Comment by riccardomurri** [comment]
> dbus package missing on debian 8 vanilla distribution. I added

I think this is fixed in the latest "master" code, in commit 1550c2e59779ef3622f827763ec57bbb2c89c56d -- can you please check?

**Comment by riccardomurri** [comment]
> there are many places where the playbook directive is_debian_8_or_later seems to be inconsistent.

This too should be fixed in the latest "master", commit e6ec47eb0e8815394a3f36b0faff09205aaaa905 -- again, can you please try it?

**Comment by lila** [comment]
i didn't mean to close the issue.  i will reopen and check the latest builds.

**Comment by riccardomurri** [comment]
Should be fixed in PR #492 -- can you please check?

**Comment by riccardomurri** [comment]
PR #492 merged into "master", I think this fixes the last of the reported issues.  Let me know if it doesn't ;-)
